src/components/modal/modal.spec.js

```diff
@@ -202,4 +158,50 @@ describe('Modal', () => {
       });
     });
   });
+
+  describe('when the modal is open', () => {
+    let domNode;
+    let escapeKeyEvent;
+    const onCancelFn = jest.fn();
+
+    beforeEach(() => {
+      escapeKeyEvent = new KeyboardEvent('keyup', {
+        key: 'Escape',
+        which: 27,
+        bubbles: true
+      });
+      wrapper = mount(
+        <Modal open onCancel={ onCancelFn } />
+      );
+      domNode = wrapper.getDOMNode();
+      document.body.appendChild(domNode);
+    });
+
+    afterEach(() => {
+      document.body.removeChild(domNode);
+    });
+
+    describe('and the esc key is released', () => {
+      it('stopImmediatePropagation function should have been called on the event', () => {
+        jest.spyOn(escapeKeyEvent, 'stopImmediatePropagation');
+        domNode.dispatchEvent(escapeKeyEvent);
+        expect(escapeKeyEvent.stopImmediatePropagation).toHaveBeenCalled();
+      });
+
+      it('then the onCancel method should have been called', () => {
+        onCancelFn.mockReset();
+        domNode.dispatchEvent(escapeKeyEvent);
+        expect(onCancelFn).toHaveBeenCalled();
+      });
+
+      describe('with disableEscKey prop set to true', () => {
+        it('then the onCancel method should have been called', () => {
```

Comment:
I think this should say "should not have been called"

Reply:
good catch, changed